cosmos/providers/dbt/core/utils/profiles_generator.py

```diff
@@ -19,20 +20,34 @@
 logger = logging.getLogger(__name__)
 
 
-def create_default_profiles():
+def read_file_safe(file_path):
+# Makes attempts to read the file if it is locked.
+# check the version of cosmos when it was created
+    retries = 3
+    with open(file_path, "r") as f:
+        # try to read the file, if it fails, another model
+        # has likely locked it. We'll wait a moment to see
+        # if the lock gets released.
+        for _ in range(0, retries):
+            try:
+                first_line = next(f)
+                file_contents = [line for line in f]
+                break
+            except:
+                sleep(1)
+                continue
+    return (first_line, file_contents)
```

Comment:
A little hacky, but it works to make sure that multiple models running at the same time don't throw an error when reading from a file that becomes locked (creating a StopIteration error or something like that).